lib/cdo/secrets.rb

```diff
@@ -0,0 +1,61 @@
+require 'aws-sdk-secretsmanager'
+require 'cdo/cache_method'
+require 'concurrent/promises'
+
+module Cdo
+  # Interface for fetching secrets from AWS Secrets Manager.
+  class Secrets
+    using CacheMethod
+
+    CURRENT = "AWSCURRENT".freeze
+
+    def initialize(paths:, client: nil)
+      @paths = paths
+
+      # Asynchronously load {name => id} Hash of secrets from provided path prefixes.
+      @secret_names = Concurrent::Promises.future do
+        @client = client || Aws::SecretsManager::Client.new
+        secret_list = @client.list_secrets.secret_list.map(&:name)
+        @paths.map do |path|
+          secret_list.
+            select {|name| name.start_with?(path)}.
+            map {|name| [name.sub(/^#{path}/, ''), name]}.
+            to_h
+        end.inject(&:merge)
+      rescue Aws::Errors::MissingCredentialsError, Aws::SecretsManager::Errors::ServiceError
+        # Silently fail in development where AWS credentials may not be present.
```

Comment:
you read my mind, I just tweaked this to depend on a `CDO.secrets_required` value so it can be locally configurable.

I think I'd still prefer to default to `false` for `development` though, so that it doesn't add any friction to the case where AWS/SecretsManager credentials aren't being used.